js/content/store.js

```diff
@@ -3119,9 +3174,138 @@ let WishlistPageClass = (function(){
                 DynamicStore.clear();
                 location.reload();
             });
-        }, true)
+        }, true);
     }
 
+    WishlistPageClass.prototype.showModalDialog = function() {
+        // Partly copied from shared_global.js
+        ExtensionLayer.runInPageContext(`function() {
+            let deferred = new jQuery.Deferred();
+            let fnOK = () => deferred.resolve();
+
+            let Modal = _BuildDialog(
+                "${Localization.str.export_wishlist}",
+                \`${this.exportModalTemplate}\`,
+                [], fnOK);
+            deferred.always(() => Modal.Dismiss());
+
+            Modal.m_fnBackgroundClick = () => Modal.Dismiss();
+            Modal.Show();
+
+            $J("#es_export_modal input[placeholder]").each(function() {
+                $J(this).attr('size', $J(this).attr('placeholder').length);
+            });
+
+            // attach the deferred's events to the modal
+            deferred.promise(Modal);
+        }`);
+
+        document.addEventListener("click", clickListener);
```

Comment:
This is probably not required.
shared_global.js has a function named `ShowConfirmDialog` that has a confirm and a cancel button already built in

Reply:
I don't really understand what you mean with this.

Reply:
Well, this `click` listener you're registering here, that handles the buttons, is superfluous, since there exists a function that simplifies that.
```
/**	Show a popup dialog like confirm(), with two buttons.  Clicking ok resolves with done(), cancel or closing the window resolves with fail()
 *
 * @param strTitle			Title bar text
 * @param strDescription	Message text
 * @param strOKButton		Text to show on OK button (default "OK")
 * @param strCancelButton	Text to show on Cancel button (default "Cancel")
 * @param strSecondaryActionButton	Add a secondary ok button (three buttons total).  Resolves with done() like OK but passes 'SECONDARY' as argument to handler
 * @returns CModal
 */
function ShowConfirmDialog( strTitle, strDescription, strOKButton, strCancelButton, strSecondaryActionButton )
```

Reply:
How do I call `exportWishlist` then?

Reply:
I think I get it: you want me to replace the entire existing modal code with ShowConfirmDialog. I can agree, that would be better.

Reply:
Reading the source of Steam's JS files really helps here.
You can call the `done(callback)` method on the returned object.

Otherwise, there are two calls to this function in out codebase, for reference

Reply:
Ok, I came across a problem. After done(), the modal content is no longer available, hence I cannot retrieve the options.

Reply:
Okay, I resolved it by listening for a form change and updating the options object on the fly.